server/src/main/java/org/apache/uniffle/server/buffer/ShuffleBuffer.java

```diff
@@ -47,6 +47,8 @@ public class ShuffleBuffer {
 
   private final long capacity;
   private long size;
+  // for Netty mode
+  private long estimatedSize;
```

Comment:
Do we need estimatedSize? Could we reuse estimatedSize?

Reply:
Because we use the accurate real-time used direct memory `pinnedDirectMemory` to determine whether to do the pre-allocation(or flush) or not. If we use `size` to calculate `usedMemory`, `usedMemory` will gradually deviate from `pinnedDirectMemory` over time. This will lead to inaccuracies when calling the `pickFlushedShuffle` method and when the coordinator allocates shuffle servers, as they both continue to use `usedMemory` as the basis for judgment.

Reply:
And also, we cannot reuse `size`, because the real size of file will be used in places like: 
`LocalStorageManager`.`updateWriteMetrics`
`HadoopStorageManager`.`updateWriteMetrics`
`ShuffleTaskInfo`.`addOnLocalFileDataSize`
`ShuffleTaskInfo`.`addOnHadoopDataSize`
`StorageWriteMetrics`.`eventSize`

We have to keep it the original way.